mcp23017_driver/mcp23017.c

```diff
@@ -0,0 +1,593 @@
+#include "mcp23017.h"
+
+typedef struct {
+    uint8_t IODIRA,
+    IODIRB,
+    IPOLA,
+    IPOLB,
+    GPINTENA,
+    GPINTENB,
+    DEFVALA,
+    DEFVALB,
+    INTCONA,
+    INTCONB,
+    IOCON,
+    GPPUA,
+    GPPUB,
+    INTFA,
+    INTFB,
+    INTCAPA,
+    INTCAPB,
+    _GPIOA,
+    _GPIOB,
+    OLATA,
+    OLATB;
+} Registers_t;
+
+Registers_t bank0_registers_addresses = {
```

Comment:
You can assume only one configuration of bank registers (the one with IOBANK at default state). Nobody will never use the other one and nobody knows why that register exists.

Reply:
Removed the code that deals with register management in case of different BANK configuration.
Now the code assumes only BANK = 0 mode which is the default one.